src/app.js

```diff
@@ -6,8 +6,193 @@ import _ from 'underscore';
+
+  tripList.on('update', renderAll);
+  tripList.on('sort', renderAll);
+
+  tripList.forEach((trip) => {
+    let tripsHTML = allTripsTemplate(trip.attributes);
+    if (trip.id < 34) {
+      tripListElement.append( $(tripsHTML) );
+    }
+  }); // for each
+}; // renderAll function
+
+// error messages
+
+const reportStatus = function(status, message) {
+  console.log(`Reporting ${ status } message: ${ message } `);
+  const generatedHTML = statusMessageTemplate({
+    status: status,
+    message: message,
+  });
+
+  $('#status-messsages ul').append(generatedHTML);
+  $('status-messages').show()
+}
+
+const tripsApiErrorHandler = function(model, response) {
+  model.destroy();
+
+  if (response.responseJSON['errors']) {
+    const errors = response.responseJSON['errors'];
+    for (const field in errors) {
+      for (const problem of errors[field]) {
+        reportStatus('error', ` ${ field }: ${ problem }`);
+      } // for problem
+    } // error
+
+  } else {
+   reportStatus('error', 'Could not save trip');
+   console.log('Error response from server:')
+   console.log(response);
+ } // end else
+} //handler
+
+const renderDetails = function renderDetails(id) {
+  const tripDetails = $('#trip-details');
+  tripDetails.empty();
+  const oneTrip = new Trip({id: id});
+  oneTrip.fetch({}).done(() => {
+    tripDetails.append(showDetailsTemplate(oneTrip.attributes));
+  });
+}; // renderDetails
+
+const makeReservation = function makeReservation(bookTripID, reservationFormData) {
+  let reserveURL = 'https://trektravel.herokuapp.com/trips/' + bookTripID + '/reservations';
+  $.post(reserveURL, reservationFormData);
+}
+
+const readAddTripForm = function readAddTripForm() {
+  const addTripData = {};
+  TRIP_FIELDS.forEach( (field) => {
+    const inputElement = $(`add-trip-form input[name="${ field }"]`);
+    addTripData[field] = inputElement.val();
+  });
+  return addTripData;
+}; // read form function
+
+
+const clearForm = function clearForm() {
+  $('#add-trip-form input[name]').val('');
+};
+
+
+// Document Ready
 $(document).ready( () => {
-  $('main').html('<h1>Hello World!</h1>');
-});
+
+
+  allTripsTemplate = _.template($('#all-trips-template').html() );
+  showDetailsTemplate = _.template($('#show-details-template').html() );
+  statusMessageTemplate = _.template($('#status-message-template').html() );
+
+  $('#individual-trip').hide();
+  $('#all-trips').hide();
+  $('#reserve-trip').hide();
+  $('#add-trip').hide();
+  $('#status-messages').hide();
+
+  const tripList = new TripList();
+
+  tripList.on('sort', renderAll);
+
+  //Events
+
+  // show add trip form
+
+  $('#create-trip').on('click', function() {
+    console.log("You clicked to see the add trip form");
+    $('#add-trip').show();
+  });
+
+  // add a new trip
+
+  $('#add-trip-form').on('submit', (event) => {
+    event.preventDefault();
+
+    $('#individual-trip').hide();
+    const newTrip = new Trip(readAddTripForm);
+    tripList.add(newTrip);
```

Comment:
no checking for validations?

Also the trip doesn't have an ID yet (gets the from the API), so you **should** wait until the save is successful and then execute `tripList.fetch()` to pull down the updated Trip from the API (with an ID).